### PR TITLE
Interactive mode, save flag and code refactor

### LINKED ISSUE #4: Interactive version with enquirer
Hi guys,

what do you think about an interactive version with https://www.npmjs.com/package/enquirer?

We can guide the user throghout specific flow with standard placeholders, that are the same default values like

days -> 30
branch -> main

### LINKED ISSUE #5: Show the heatmap visually or save the json
What do you think about an optional argument that let the user to show the heatmap visually ( via a specific frontend that show the generated json visually with the help of library like d3.js ).

And the --save argument, that save the .json directly in a specific folder, that let the users to use it in other contexts like CI.

### PR DESCRIPTION
**Pull request by buggyzap**
This PR add the following features:

- Users can run the command without args and get in interactive mode. They got prompted all required fields to execute spyone.
- --save flag let the user to execute the tool in CI context, or automation scripts in custom flows. This features let the user to save the file without prompt nothing and without starting the server. In this case we except that the command will be launched from a machine.
- Standardized help

The code refactor clarify the flow of the args thanks to the new args-parser library.
We are able to execute the tool in both flows, interactive mode and automated task, and the variables will be overriden to the specific flow.

I also added a new help.mjs file, this file help us to write a standalone help text that will be showed in case the user launch
the --help, -help, or just help.

This help get showed to the user also in case they type an unexpected options.

Feedback and suggestions are also appreciated.

### CODE REVIEW

**Comment by Cadienvan** [comment]
I really love it, it's pretty powerful tbh!

My only gripe is related to the arguments being duplicated.
Do you think it could be plausible to bring them all in a single array and proceed from there? I mean.. arguments for Parse and Enquirer are the same, both in naming and format, do you see it viable? I'd merge it in any case if the rest of the team agrees, just wanted to hear your opinion!

**Comment by buggyzap** [comment]
> I really love it, it's pretty powerful tbh!
> 
> My only gripe is related to the arguments being duplicated. Do you think it could be plausible to bring them all in a single array and proceed from there? I mean.. arguments for Parse and Enquirer are the same, both in naming and format, do you see it viable? I'd merge it in any case if the rest of the team agrees, just wanted to hear your opinion!

Hi @Cadienvan , I don't undertstand your proposal, can you do a little suggest directly in the code or explain me a bit much more?

Now we have some variables in the scope, that are creating from Parser ( cli ) and fallback to default values.
Next we have a prompt array, with the Prompt structure.
Next we check that if no args are provided, we initialize the Prompt or we go directly down on the code.
In case we initialize the prompt, the previous variables got overriden from the Prompt.

In this flow, what do you suggest? To create a top-level array with all arguments and adapt them on the specific scenario? Or take the prompt array as "Master" and iterate his args to create the default one in case no args are provided? 😮

**Comment by Cadienvan** [comment]
Hey @buggyzap I mean your first suggestion, bringing both configurations under a master array so that the maintenance becomes strictly related to a specific portion of the code. Do you see any drawback or hard steps to achieve it?

As I said earlier, I'd bring this PR in regardless, it's just an exchange of opinions.

**Comment by buggyzap** [comment]
> Hey @buggyzap I mean your first suggestion, bringing both configurations under a master array so that the maintenance becomes strictly related to a specific portion of the code. Do you see any drawback or hard steps to achieve it?
> 
> As I said earlier, I'd bring this PR in regardless, it's just an exchange of opinions.

I agree, now both implementations take the **cliArguments** object as interface to correspective implementations.

I also added a little adjustment to Enquirer array transformation. 

Please check the code and give me a feedback if this option is suitable and better maintenaible also for you.

File: index.mjs [hunk 1]
```diff
@@ -87,12 +150,20 @@ const stats = [...data.entries()].reduce((prev, curr) => {
...
+  process.exit(0);
+}
+
 const server = http.createServer(function (req, res) {
   fs.readFile(resultsFilePath, function (err, data) {
     if (err) throw err;
 
-    if (outputFormat !== 'json' && outputFormat !== 'html') {
-      const errorMessage = '❌ Output format not supported (json or html)';
+    if (!cliArguments.output.choices.includes(output)) {
```
Comment (Cadienvan) [inline index.mjs]: This doesn't work correctly.

If you use the interactive version, cliArguments.output.choices is an array of objects.
If you use the parametrized version, this is the array you expect.

**Review by Cadienvan — CHANGES_REQUESTED** [review]

File: index.mjs (hunk 1 above)
Comment (buggyzap) [reply index.mjs]: Sure thanks

**Comment by buggyzap** [comment]
> I love it, I just added a change request because depending on the version you call, the HTTP server fails because of a different data structure! When that is gone (Even just with a simple comment and an "if" checking the type) we can merge it!

Ok @Cadienvan I think now is done

**Review by Cadienvan — APPROVED** [review]

**Review by PellegrinoDurante — APPROVED** [review]

**Pull request merged**